backend/open_webui/retrieval/loaders/datalab_marker.py

```
@@ -0,0 +1,278 @@
+                response = requests.post(
+                    f"{self.api_base_url}/marker",
+                    data=form_data,
+                    files=files,
+                    headers=headers,
+                )
+                response.raise_for_status()
+                result = response.json()
+        except FileNotFoundError:
+            raise HTTPException(
+                status.HTTP_404_NOT_FOUND, detail=f"File not found: {self.file_path}"
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)